### PR TITLE
fix(material/datepicker): display datepicker inside a dialog example

### LINKED ISSUE #28186: bug(angular material datepicker): datepicker throws an    NullInjectorError: No provider for DateAdapter! that is not documented in any example
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

13.0.0

### Description

I want to use angular material datepicker for date range selection in my application. This worked just in previous versions I used to work with.
I follow the documentation and yet the following error rises: 
create-trip.component.html:10 ERROR NullInjectorError: R3InjectorError(Standalone[_HomeComponent])[DateAdapter -> DateAdapter -> DateAdapter -> DateAdapter]: 
  NullInjectorError: No provider for DateAdapter! 

The datepicker is located inside an angular material dialog component, which is opened from the HomeComponent that is mentioned in the error. I tried to provide 'NativeDateAdapter' in both the dialog component and the home component but it didn't help.

Just to clarify, the calendar doesn't open and I can't choose dates.

### Reproduction

StackBlitz link: 
Steps to reproduce:
1. create a dialog component
2. place a datepicker inside it
3.  try to choose a date

### Expected Behavior

When the calendar icon is clicked, a calendar opens.

### Actual Behavior

the calendar doesn't open and the mentioned error is shown in the console.

### Environment

Angular CLI: 17.0.0
Node: 20.9.0
Package Manager: npm 10.1.0
OS: win32 x64

Angular: 17.0.2
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router   

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.0
@angular-devkit/build-angular   17.0.0
@angular-devkit/core            17.0.0
@angular-devkit/schematics      17.0.0
@angular/cdk                    17.0.0
@angular/cli                    17.0.0
@angular/google-maps            17.0.0
@angular/material               17.0.0
@schematics/angular             17.0.0
rxjs                            7.8.1
typescript                      5.2.2
zone.js                         0.14.2

### PR DESCRIPTION
**Pull request by GiftLanga**
Adds an example that demonstrates how to display a datepicker inside a matDialog component.

Fixes #28186

### CODE REVIEW

**Comment by Totati** [comment]
Maybe it would worth mentioning that the the `open` method accetps an injector, so it can use the providers of the opening component as well. 🙂

File: src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example-dialog.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<h2 mat-dialog-title>Calendar in a Dialog</h2>
+<mat-dialog-content align="center">
+  <div class="example-calendar-container">
+    <mat-calendar [(selected)]="data.selectedDate"></mat-calendar>
```
Comment (crisbeto) [inline src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example-dialog.html:4]: This is only showing a calendar inside the dialog, not a full datepicker. Should we rename the example?

File: src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.example-calendar-container {
```
Comment (crisbeto) [inline src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example.css:1]: Could you avoid these styles by passing the width/height into the dialog `open` call?

Comment (GiftLanga) [reply src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example.css]: Updated

File: src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example-dialog.html (hunk 1 above)
Comment (GiftLanga) [reply src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example-dialog.html]: I updated it to use the datepicker instead.

File: src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  selector: 'datepicker-dialog-example',
+  templateUrl: 'datepicker-dialog-example-dialog.html',
+  standalone: true,
+  imports: [
+    MatDatepickerModule,
+    MatDialogModule,
+    MatButtonModule,
+    MatFormFieldModule,
+    MatInputModule,
+    FormsModule,
```
Comment (Totati) [inline src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example.ts:53]:
```suggestion
```
It's unnecessary.

File: src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example-dialog.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+<mat-dialog-content align="center">
+  <mat-form-field>
+    <mat-label>Select a date</mat-label>
+    <input matInput [matDatepicker]="picker" [formControl]="date">
+    <mat-datepicker-toggle matIconSuffix [for]="picker"></mat-datepicker-toggle>
+    <mat-datepicker #picker></mat-datepicker>
+  </mat-form-field>
+</mat-dialog-content>
+<mat-dialog-actions>
+  <button mat-button (click)="onClear()">Clear</button>
```
Comment (Totati) [inline src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example-dialog.html:11]: I see that this example is based on the "Dialog Overview", however using `mat-dialog-close` should reduce the unrelated code even more.
```suggestion
  <button mat-button mat-dialog-close>Clear</button>
```
This would eliminate the need for the `onClear` method.

File: src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example.ts (hunk 3 above)
Comment (GiftLanga) [reply src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example.ts]: Updated

File: src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example-dialog.html (hunk 4 above)
Comment (GiftLanga) [reply src/components-examples/material/datepicker/datepicker-dialog/datepicker-dialog-example-dialog.html]: Thanks, I updated it.

**Review by crisbeto — APPROVED** [review]
LGTM, but there's a lint warning that's failing the CI.
